site/content/en/concepts/setters/_index.md

```diff
@@ -0,0 +1,84 @@
+---
+title: "Setters"
+linkTitle: "Setters"
+weight: 4
+type: docs
+description: >
+  Setters and related concepts
+---
+
+## Setters
+
+Setters provide a solution for template-free setting or substitution of field
+values through package metadata (OpenAPI). They are a safer alternative to other
+substitution techniques which do not have the context of the structured data
+– e.g. using `sed` to replace values.
+
+The OpenAPI definitions for setters are defined in a Kptfile and referenced by a
+fields through comments on the fields.
+
+Setters may be invoked to programmatically modify the configuration using `kpt cfg set`
```

Comment:
I'd drop "programatically" here. The interface exposed to user is not programatic.